### PR TITLE
Revert vig and odd stats changes

### PR DESCRIPTION
**Pull request by veloci-raptor**
## Description

---

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Added Features

- [ ]

---

## Modified Features

- [ ]

---

## Test Cases

- [ ] Test A
- [ ] Test B

---

## Change Log

---

## Documentation

---

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
---

### CODE REVIEW

File: proto/sge/sportevent/params.proto [hunk 1]
```diff
@@ -38,16 +38,4 @@ message Params {
     (gogoproto.nullable)   = false
   ];
 
-  string event_max_vig = 5 [
```
Comment (3eyed-raven) [inline proto/sge/sportevent/params.proto:41]: event_max_loss : This field is not used anywhere. Please remove it.

File: proto/sge/sportevent/sport_event.proto [hunk 2]
```diff
@@ -53,35 +53,7 @@ message EventBetConstraints {
...
-    (gogoproto.nullable)   = false];
-}
-
-message TotalStats {
-  string house_loss = 1[
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
-    (gogoproto.nullable)   = false];
-
-  string bet_amount = 2[
+  string current_total_amount = 5[
```
Comment (3eyed-raven) [inline proto/sge/sportevent/sport_event.proto:56]: This field is also not required

File: x/bet/keeper/place_bet.go [hunk 3]
```diff
@@ -26,33 +26,18 @@ func (k Keeper) PlaceBet(ctx sdk.Context, bet *types.Bet, activeBetOdds []*types
...
-	}
-
 	// modify the bet fee and subtracted amount
 	setBetFee(bet, sportEvent.BetConstraints.BetFee)
 
 	// calculate extraPayout
 	extraPayout := calculateExtraPayout(bet)
 
-	if err := k.sporteventKeeper.AddExtraPayoutToEvent(ctx, sportEvent, bet.OddsUID, bet.Amount, extraPayout); err != nil {
+	if err := k.sporteventKeeper.AddExtraPayoutToEvent(ctx, sportEvent.UID, extraPayout); err != nil {
```
Comment (3eyed-raven) [inline x/bet/keeper/place_bet.go:40]: Do we need to add extra payout to event?

File: x/bet/types/errors.go [hunk 4]
```diff
@@ -51,7 +51,6 @@ var (
 	ErrOddsMaxLossExceeded        = sdkerrors.Register(ModuleName, 2041, "odds loss is in max state")
 	ErrOddsDataNotFound           = sdkerrors.Register(ModuleName, 2042, "odds not exist in ticket payload")
 	ErrInvalidOddsType            = sdkerrors.Register(ModuleName, 2043, "valid odds type should be provided, 1 for Decimal, 2 for Fractional, 3 for American")
-	ErrActiveOddsUIDsNotValid     = sdkerrors.Register(ModuleName, 2044, "all provided active odds UIDs in ticket should be related to the same sport event and valid")
 	ErrUserKycFailed              = sdkerrors.Register(ModuleName, 2045, "KYC Validation failed the bettor")
```
Comment (3eyed-raven) [inline x/bet/types/errors.go:54]: ErrUserKycFailed --> change number from 2045 to 2044 because you removed 2044

File: x/bet/types/errors.go [hunk 5]
```diff
@@ -51,7 +51,6 @@ var (
...
 	ErrTextInvalidCreator         = "invalid creator address (%s)"
```
Comment (3eyed-raven) [inline x/bet/types/errors.go:55]: Why this error does not follow the same error format as all other errors?

File: x/sportevent/keeper/msg_server_add_event.go [hunk 6]
```diff
@@ -83,17 +83,10 @@ func initBetConstraints(event *types.SportEvent, params types.Params) bool {
 	// init all Bet constraints if nil
 	if event.BetConstraints == nil {
 		event.BetConstraints = &types.EventBetConstraints{
-			MaxBetCap:      params.EventMaxBetCap,
```
Comment (3eyed-raven) [inline x/sportevent/keeper/msg_server_add_event.go:86]: max bet cap, min amount, betfee constraints should not be removed
??

File: x/sportevent/keeper/msg_server_update_event.go [hunk 7]
```diff
@@ -108,12 +104,6 @@ func (k msgServer) validateEventUpdate(ctx sdk.Context, event, previousEvent typ
 	if event.BetConstraints.MaxLoss.GT(params.EventMaxLoss) {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "max loss cannot be greater than the systems limit")
 	}
-	if event.BetConstraints.MaxVig.GT(params.EventMaxVig) {
```
Comment (3eyed-raven) [inline x/sportevent/keeper/msg_server_update_event.go:111]: Please remove time.Now() checks 
use ctx.Blocktime

File: x/sportevent/keeper/sport_event.go [hunk 8]
```diff
@@ -81,37 +77,20 @@ func (k Keeper) ResolveSportEvents(ctx sdk.Context, resolutionEvent *types.Resol
...
+	sportEvent, found := k.GetSportEvent(ctx, sportEventUID)
+	if !found {
+		return types.ErrNoMatchingSportEvent
 	}
 
-	// update new total statistics
-	sportEvent.BetConstraints.TotalStats.BetAmount = sportEvent.BetConstraints.TotalStats.BetAmount.Add(betAmount)
-	if sportEvent.BetConstraints.TotalStats.BetAmount.GT(sportEvent.BetConstraints.MaxBetCap) {
+	// calculate new total payout
+	newAmount := sportEvent.BetConstraints.CurrentTotalAmount.Add(amount)
```
Comment (3eyed-raven) [inline x/sportevent/keeper/sport_event.go:87]: this code block will not be required. Right?

File: x/sportevent/types/params.go [hunk 9]
```diff
@@ -20,12 +20,6 @@ var (
...
 	KeyMinBetFee = []byte("EventMinBetFee")
 
-	// KeyMinBetFee is the minimum bet fee param key
-	KeyMinVig = []byte("KeyMinVig")
-
-	// KeyMinBetFee is the minimum bet fee param key
-	KeyMaxVig = []byte("KeyMaxVig")
-
 	// KeyMinBetFee is the minimum bet fee param key
 	KeyMaxEventLoss = []byte("KeyMaxEventLoss")
```
Comment (3eyed-raven) [inline x/sportevent/types/params.go:24]: remove this

File: x/sportevent/types/params.go [hunk 10]
```diff
@@ -40,12 +34,6 @@ const (
 
 	// DefaultMaxEventLoss is the default max loss event can have
 	DefaultMaxEventLoss = 1000000000000
```
Comment (3eyed-raven) [inline x/sportevent/types/params.go:36]: this too

File: x/sportevent/types/params.go [hunk 11]
```diff
@@ -81,8 +67,6 @@ func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
 		paramtypes.NewParamSetPair(KeyEventMinBetAmount, &p.EventMinBetAmount, validateMinBetAmount),
 		paramtypes.NewParamSetPair(KeyMaxBetCap, &p.EventMaxBetCap, validateMaxBetCap),
 		paramtypes.NewParamSetPair(KeyMinBetFee, &p.EventMinBetFee, validateMinBetFeePercentage),
-		paramtypes.NewParamSetPair(KeyMinVig, &p.EventMinVig, validateMinVig),
-		paramtypes.NewParamSetPair(KeyMaxVig, &p.EventMaxVig, validateMaxVig),
 		paramtypes.NewParamSetPair(KeyMaxEventLoss, &p.EventMaxLoss, validateMaxEventLoss),
```
Comment (3eyed-raven) [inline x/sportevent/types/params.go:70]: this too

File: x/sportevent/types/params.go [hunk 12]
```diff
@@ -64,8 +52,6 @@ func NewParams() Params {
 		EventMinBetAmount: sdk.NewInt(DefaultMinBetAmount),
 		EventMaxBetCap:    sdk.NewInt(DefaultMaxBetCap),
 		EventMinBetFee:    DefaultMinBetFee,
-		EventMaxVig:       sdk.NewDec(DefaultMaxVig),
-		EventMinVig:       sdk.NewDec(DefaultMinVig),
 		EventMaxLoss:      sdk.NewInt(DefaultMaxEventLoss),
```
Comment (3eyed-raven) [inline x/sportevent/types/params.go:55]: this too

File: proto/sge/sportevent/params.proto (hunk 1 above)
Comment (veloci-raptor) [reply proto/sge/sportevent/params.proto:41]: done

File: proto/sge/sportevent/sport_event.proto (hunk 2 above)
Comment (veloci-raptor) [reply proto/sge/sportevent/sport_event.proto:56]: done

File: x/bet/keeper/place_bet.go (hunk 3 above)
Comment (veloci-raptor) [reply x/bet/keeper/place_bet.go:40]: removed

File: x/bet/types/errors.go (hunk 4 above)
Comment (veloci-raptor) [reply x/bet/types/errors.go:54]: done

File: x/bet/types/errors.go (hunk 5 above)
Comment (veloci-raptor) [reply x/bet/types/errors.go:55]: discussed

File: x/sportevent/keeper/msg_server_add_event.go (hunk 6 above)
Comment (veloci-raptor) [reply x/sportevent/keeper/msg_server_add_event.go:86]: removed

File: x/sportevent/keeper/sport_event.go [hunk 13]
```diff
@@ -81,37 +77,20 @@ func (k Keeper) ResolveSportEvents(ctx sdk.Context, resolutionEvent *types.Resol
...
-			ExtraPayout: extraPayout,
-		}
-	}
-	sportEvent.BetConstraints.TotalOddsStats[oddsUID] = stats
-
-	// set house loss statistics
-	sportEvent.BetConstraints.TotalStats.HouseLoss = houseLoss
-
+	// update bet constraints of sport event in module state
+	sportEvent.BetConstraints.CurrentTotalAmount = newAmount
```
[the remark this answers is not in the document]
Comment (veloci-raptor) [reply x/sportevent/keeper/sport_event.go:93]: removed

File: x/sportevent/types/params.go (hunk 9 above)
Comment (veloci-raptor) [reply x/sportevent/types/params.go:24]: removed

File: x/sportevent/types/params.go (hunk 10 above)
Comment (veloci-raptor) [reply x/sportevent/types/params.go:36]: removed

**Review by 3eyed-raven — APPROVED** [review]

**Pull request merged**